src/SFA.DAS.AssessorService.EpaoImporter/DomainServices/PrintingSpreadsheetCreator.cs

```diff
@@ -163,40 +163,38 @@ private static void CreateWorksheetHeader(int batchNumber, ExcelWorksheet worksh
 
         private static string NameToTitleCase(string name)
         {
-            var isAllLower = true;
-            var isAllUpper = true;
-            
-            foreach (var character in name)
-            {
-                if (char.IsLetter(character) &&  !char.IsUpper(character))
-                {
-                    isAllUpper = false;
-                }
-            }
-            
-            foreach (var character in name)
+            var convertToTitleCase = IsNameAllUpperOrAllLower(name);
+
+            if (!convertToTitleCase) return name;
+            var newNameCharacters = CultureInfo.CurrentCulture.TextInfo.ToTitleCase(name.ToLower()).ToCharArray();
+
+            FixTitleCaseForNamesWithApostrophesAndHyphens(newNameCharacters);
+            return new string(newNameCharacters);
+        }
+
+        private static bool IsNameAllUpperOrAllLower(string name)
+        {
+            var convertToTitleCase = name.ToCharArray().Where(char.IsLetter).All(char.IsLower);
+            if (!convertToTitleCase)
             {
-                if (char.IsLetter(character) &&  
-                    char.IsUpper(character))
-                {
-                    isAllLower = false;
-                }
+                convertToTitleCase = name.ToCharArray().Where(char.IsLetter).All(char.IsUpper);
             }
 
-            if (!isAllUpper && !isAllLower) return name;
-            var chars = CultureInfo.CurrentCulture.TextInfo.ToTitleCase(name.ToLower()).ToCharArray();
+            return convertToTitleCase;
+        }
 
-            for (var i = 0; i + 1 < chars.Length; i++)
+        private static void FixTitleCaseForNamesWithApostrophesAndHyphens(char[] newNameCharacters)
+        {
+            for (var i = 0; i + 1 < newNameCharacters.Length; i++)
             {
-                if (chars[i].Equals('\'') ||
-                    chars[i].Equals('`') ||
-                    chars[i].Equals('-'))
-                {                    
-                    chars[i + 1] = char.ToUpper(chars[i + 1]);
```

Comment:
This is probably extra fussy, but this will break if any of those symbols (quote, hyphen) appear as last character in string.  I'm not going to no approve based on that, but as you're closer to domain issue, please evaluate and fix if needed.

Reply:
Bloody good point.  👍